Stratis.Bitcoin/Features/MemoryPool/MempoolManager.cs

```diff
@@ -245,5 +253,18 @@ public Task RemoveForBlock(Block block, int blockHeight)
                 this.Validator.PerformanceCounter.SetMempoolDynamicSize(this.memPool.DynamicMemoryUsage());
             });
         }
+
+        /// <inheritdoc />
+        public async Task<UnspentOutputs> GetUnspentTransactionAsync(uint256 trxid)
+        {
+            var txInfo = await InfoAsync(trxid);
+            if(txInfo == null)
+            {
+                return null;
+            }
+            var memPoolCoinView = new MempoolCoinView(this.coinView, this.memPool, this.MempoolLock, this.Validator);
```

Comment:
I think if you use MempoolCoinView you dont need the call before that InfoAsync()

Reply:
I was doing that to get a reference to the transaction to feed in to the load method. Is there a way to load the unspent coins with just using the hash?

Reply:
ah yeah good point, come to think about it if a trx is in the mempool its outputs are probably not spent (unless its parent is in the mempool) and that will be detected by the MempoolCoinView.
So I guess you are right we want to see is the trx parents are not in the memppol so this is the best approach (else we would just return the given trx).

Reply:
MempoolCoinView exposes a "HasCoins" method, but that doesn't grab the transaction from memory. It simply checks if it exists. The alternative is to pull some of the logic out of MempoolCoinView.LoadView and just use that in the manager. Otherwise, I think I have to leave this as is.

Reply:
yes leave it as is